teps/0004-tekton-integrations-annotations-statuses.md

```diff
@@ -0,0 +1,739 @@
+- Chat tools (Slack, Hangouts, etc.)
+- Email
+- Bug trackers (GitHub, Jira, etc.)
+- Webhook / Cloud Event
+
+Currently most integration solutions for Tekton Pipelines require users to embed
+update Tasks into their Pipelines. In practice, this ends up not being a good
+mechanism for ensuring that external integration notifications always occur, for
+example:
+
+- If the pipeline fails to start because of a configuration error, the external
+  integration is never notified.
+- If the pipeline fails a step and never reaches the notification step, the
+  external integration is never notified.
+- If a transient error (e.g. GitHub is down) prevents the task from completing
+  successfully, this may result in the pipeline failing.
+
+This proposal details improvements to Task/PipelineRuns to annotate and store
+integration status information.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Define a mechanism for Tekton and other providers to hook in external
+  integrations for pipeline and task runs.
+- Allow integrations to report integration status for a pipeline for ease of
+  debugging.
+- Minimize impact on pipeline/task execution spec.
+
+<!--
+List the specific goals of the TEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+<!--
+What is out of scope for this TEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- While GitHub integrations will often be used as an example, we want to avoid
+  going into too much deep details into how a specific GitHub integration will
+  be configured / what data will be rendered / what APIs will be used / etc,
+  unless it has direct implications on integrations as a whole.
+
+## Requirements
+
+<!--
+List the requirements for this TEP.
+-->
+
+1. Integrations should be optional from core Tekton Pipelines functionality.
+2. Users should be able to pick and choose which integrations they have
+   installed in their cluster.
+3. Integration data should be stored as close to the PipelineRun as possible
+   (ideally alongside).
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Integration Annotations
+
+[Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
+of the format `<name>.integrations.tekton.dev/<value>` to will be used to
```

Comment:
do you mean : `<name>.integrations.tekton.dev/<key>: <value>`

Reply:
Yup! I ended up removing this section though, so nothing to do. 😅